0.92.x fails running integration on at least java 17 and 18

**Describe the bug**
since 0.92 with external integration runs done in #1290 it now fails on java 17 and 18 both when running internally or externally. 

**To Reproduce**
Steps to reproduce the behavior:
1. `jbang init -t qrest q.java` 
2. have java 11 as default and run `jbang --fresh --java 17 q.java`

```
jbang --fresh --java 17 test.java
[jbang] Resolving dependencies...
[jbang] Artifacts used for dependency management:
         io.quarkus:quarkus-bom:pom:1.11.0.Final
[jbang] io.quarkus:quarkus-resteasy
Done
[jbang] Dependencies resolved
[jbang] Building jar...
[jbang] Running external post build for io.quarkus.launcher.JBangIntegration
[jbang] [ERROR] Issue running postBuild()
[jbang] Run with --verbose for more details
Exception in thread "main" com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1 column 54 path $.integration
	at com.google.gson.Gson.fromJson(Gson.java:942)
	at com.google.gson.Gson.fromJson(Gson.java:870)
	at dev.jbang.spi.IntegrationManager.main(IntegrationManager.java:269)
Caused by: java.io.EOFException: End of input at line 1 column 54 path $.integration
	at com.google.gson.stream.JsonReader.nextNonWhitespace(JsonReader.java:1397)
	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:483)
	at com.google.gson.stream.JsonReader.hasNext(JsonReader.java:415)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:216)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	... 2 more
```

weird one is that when running with verbose you get another error:

```
jbang --verbose --fresh --java 17 test.java
[jbang] jbang version 0.92.1
[jbang] Building as fresh build explicitly requested.
[jbang] System Java version detected as 17
[jbang] System Java version matches requested version 17
[jbang] Resolving dependencies...
[jbang] Artifacts used for dependency management:
         io.quarkus:quarkus-bom:pom:1.11.0.Final
[jbang] io.quarkus:quarkus-resteasy
Done
[jbang] Dependencies resolved
[jbang] Building jar...
[jbang] compile: /Users/max/.sdkman/candidates/java/current/bin/javac -parameters -classpath /Users/max/.m2/repository/io/quarkus/quarkus-resteasy/1.11.0.Final/quarkus-resteasy-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/quarkus-vertx-http/1.11.0.Final/quarkus-vertx-http-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/quarkus-core/1.11.0.Final/quarkus-core-1.11.0.Final.jar:/Users/max/.m2/repository/jakarta/annotation/jakarta.annotation-api/1.3.5/jakarta.annotation-api-1.3.5.jar:/Users/max/.m2/repository/jakarta/enterprise/jakarta.enterprise.cdi-api/2.0.2/jakarta.enterprise.cdi-api-2.0.2.jar:/Users/max/.m2/repository/jakarta/el/jakarta.el-api/3.0.3/jakarta.el-api-3.0.3.jar:/Users/max/.m2/repository/jakarta/interceptor/jakarta.interceptor-api/1.2.5/jakarta.interceptor-api-1.2.5.jar:/Users/max/.m2/repository/jakarta/inject/jakarta.inject-api/1.0/jakarta.inject-api-1.0.jar:/Users/max/.m2/repository/io/quarkus/quarkus-ide-launcher/1.11.0.Final/quarkus-ide-launcher-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/quarkus-development-mode-spi/1.11.0.Final/quarkus-development-mode-spi-1.11.0.Final.jar:/Users/max/.m2/repository/io/smallrye/config/smallrye-config/1.10.0/smallrye-config-1.10.0.jar:/Users/max/.m2/repository/io/smallrye/common/smallrye-common-annotation/1.5.0/smallrye-common-annotation-1.5.0.jar:/Users/max/.m2/repository/io/smallrye/config/smallrye-config-common/1.10.0/smallrye-config-common-1.10.0.jar:/Users/max/.m2/repository/io/smallrye/common/smallrye-common-expression/1.5.0/smallrye-common-expression-1.5.0.jar:/Users/max/.m2/repository/io/smallrye/common/smallrye-common-function/1.5.0/smallrye-common-function-1.5.0.jar:/Users/max/.m2/repository/io/smallrye/common/smallrye-common-constraint/1.5.0/smallrye-common-constraint-1.5.0.jar:/Users/max/.m2/repository/io/smallrye/common/smallrye-common-classloader/1.5.0/smallrye-common-classloader-1.5.0.jar:/Users/max/.m2/repository/org/jboss/logging/jboss-logging/3.4.1.Final/jboss-logging-3.4.1.Final.jar:/Users/max/.m2/repository/org/jboss/logmanager/jboss-logmanager-embedded/1.0.6/jboss-logmanager-embedded-1.0.6.jar:/Users/max/.m2/repository/org/jboss/logging/jboss-logging-annotations/2.2.0.Final/jboss-logging-annotations-2.2.0.Final.jar:/Users/max/.m2/repository/org/jboss/threads/jboss-threads/3.2.0.Final/jboss-threads-3.2.0.Final.jar:/Users/max/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar:/Users/max/.m2/repository/org/jboss/slf4j/slf4j-jboss-logmanager/1.1.0.Final/slf4j-jboss-logmanager-1.1.0.Final.jar:/Users/max/.m2/repository/org/graalvm/sdk/graal-sdk/20.3.0/graal-sdk-20.3.0.jar:/Users/max/.m2/repository/org/wildfly/common/wildfly-common/1.5.4.Final-format-001/wildfly-common-1.5.4.Final-format-001.jar:/Users/max/.m2/repository/io/quarkus/quarkus-bootstrap-runner/1.11.0.Final/quarkus-bootstrap-runner-1.11.0.Final.jar:/Users/max/.m2/repository/io/smallrye/common/smallrye-common-io/1.5.0/smallrye-common-io-1.5.0.jar:/Users/max/.m2/repository/io/quarkus/quarkus-security-runtime-spi/1.11.0.Final/quarkus-security-runtime-spi-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/quarkus-vertx-http-dev-console-runtime-spi/1.11.0.Final/quarkus-vertx-http-dev-console-runtime-spi-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/arc/arc/1.11.0.Final/arc-1.11.0.Final.jar:/Users/max/.m2/repository/jakarta/transaction/jakarta.transaction-api/1.3.3/jakarta.transaction-api-1.3.3.jar:/Users/max/.m2/repository/io/quarkus/security/quarkus-security/1.1.3.Final/quarkus-security-1.1.3.Final.jar:/Users/max/.m2/repository/io/smallrye/reactive/mutiny/0.12.5/mutiny-0.12.5.jar:/Users/max/.m2/repository/org/reactivestreams/reactive-streams/1.0.3/reactive-streams-1.0.3.jar:/Users/max/.m2/repository/io/quarkus/quarkus-vertx-core/1.11.0.Final/quarkus-vertx-core-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/quarkus-netty/1.11.0.Final/quarkus-netty-1.11.0.Final.jar:/Users/max/.m2/repository/io/netty/netty-codec/4.1.49.Final/netty-codec-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-handler/4.1.49.Final/netty-handler-4.1.49.Final.jar:/Users/max/.m2/repository/io/vertx/vertx-core/3.9.5/vertx-core-3.9.5.jar:/Users/max/.m2/repository/io/netty/netty-common/4.1.49.Final/netty-common-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-buffer/4.1.49.Final/netty-buffer-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-transport/4.1.49.Final/netty-transport-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-handler-proxy/4.1.49.Final/netty-handler-proxy-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-codec-socks/4.1.49.Final/netty-codec-socks-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-codec-http/4.1.49.Final/netty-codec-http-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-codec-http2/4.1.49.Final/netty-codec-http2-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-resolver/4.1.49.Final/netty-resolver-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-resolver-dns/4.1.49.Final/netty-resolver-dns-4.1.49.Final.jar:/Users/max/.m2/repository/io/netty/netty-codec-dns/4.1.49.Final/netty-codec-dns-4.1.49.Final.jar:/Users/max/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.11.3/jackson-core-2.11.3.jar:/Users/max/.m2/repository/io/vertx/vertx-web/3.9.5/vertx-web-3.9.5.jar:/Users/max/.m2/repository/io/vertx/vertx-web-common/3.9.5/vertx-web-common-3.9.5.jar:/Users/max/.m2/repository/io/vertx/vertx-auth-common/3.9.5/vertx-auth-common-3.9.5.jar:/Users/max/.m2/repository/io/vertx/vertx-bridge-common/3.9.5/vertx-bridge-common-3.9.5.jar:/Users/max/.m2/repository/io/quarkus/quarkus-resteasy-server-common/1.11.0.Final/quarkus-resteasy-server-common-1.11.0.Final.jar:/Users/max/.m2/repository/io/quarkus/quarkus-arc/1.11.0.Final/quarkus-arc-1.11.0.Final.jar:/Users/max/.m2/repository/org/eclipse/microprofile/context-propagation/microprofile-context-propagation-api/1.0.1/microprofile-context-propagation-api-1.0.1.jar:/Users/max/.m2/repository/io/quarkus/quarkus-resteasy-common/1.11.0.Final/quarkus-resteasy-common-1.11.0.Final.jar:/Users/max/.m2/repository/org/jboss/resteasy/resteasy-core/4.5.8.SP1/resteasy-core-4.5.8.SP1.jar:/Users/max/.m2/repository/org/jboss/spec/javax/ws/rs/jboss-jaxrs-api_2.1_spec/2.0.1.Final/jboss-jaxrs-api_2.1_spec-2.0.1.Final.jar:/Users/max/.m2/repository/org/jboss/spec/javax/xml/bind/jboss-jaxb-api_2.3_spec/2.0.0.Final/jboss-jaxb-api_2.3_spec-2.0.0.Final.jar:/Users/max/.m2/repository/org/jboss/resteasy/resteasy-core-spi/4.5.8.SP1/resteasy-core-spi-4.5.8.SP1.jar:/Users/max/.m2/repository/com/ibm/async/asyncutil/0.1.0/asyncutil-0.1.0.jar:/Users/max/.m2/repository/org/eclipse/microprofile/config/microprofile-config-api/1.4/microprofile-config-api-1.4.jar:/Users/max/.m2/repository/com/sun/activation/jakarta.activation/1.2.1/jakarta.activation-1.2.1.jar:/Users/max/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar -d /Users/max/.jbang/cache/jars/test.java.0fead24535cd346b79c14df31cc042ff76a1f9c14c1897544faa2c8213b84a2a.jar.tmp /Users/max/code/personal/jbangdev/jbang/test.java
[jbang] Running external post build for io.quarkus.launcher.JBangIntegration
[jbang] System Java version matches requested version 17
[jbang] Running: /Users/max/.sdkman/candidates/java/current/bin/java -cp /Users/max/.jbang/bin/jbang.jar dev.jbang.spi.IntegrationManager
[jbang] Deleting folder /Users/max/.jbang/cache/jars/test.java.0fead24535cd346b79c14df31cc042ff76a1f9c14c1897544faa2c8213b84a2a.jar.tmp
[jbang] [ERROR] Issue running postBuild()
dev.jbang.cli.ExitException: Issue running postBuild()
	at dev.jbang.spi.IntegrationManager.runIntegrations(IntegrationManager.java:92)
	at dev.jbang.source.builders.BaseBuilder.compile(BaseBuilder.java:164)
	at dev.jbang.source.builders.BaseBuilder.build(BaseBuilder.java:108)
	at dev.jbang.cli.BaseBuildCommand.buildIfNeeded(BaseBuildCommand.java:49)
	at dev.jbang.cli.Run.prepareArtifacts(Run.java:124)
	at dev.jbang.cli.Run.doCall(Run.java:96)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:93)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:15)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final sun.nio.fs.UnixFileSystem sun.nio.fs.UnixPath.fs accessible: module java.base does not "opens sun.nio.fs" to unnamed module @1a27aae3
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
	at com.google.gson.internal.reflect.UnsafeReflectionAccessor.makeAccessible(UnsafeReflectionAccessor.java:44)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.getBoundFields(ReflectiveTypeAdapterFactory.java:159)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory.create(ReflectiveTypeAdapterFactory.java:102)
	at com.google.gson.Gson.getAdapter(Gson.java:458)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:56)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:127)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:245)
	at com.google.gson.Gson.toJson(Gson.java:704)
	at com.google.gson.Gson.toJson(Gson.java:683)
	at com.google.gson.Gson.toJson(Gson.java:638)
	at com.google.gson.Gson.toJson(Gson.java:618)
	at dev.jbang.spi.IntegrationManager.runIntegrationExternal(IntegrationManager.java:211)
	at dev.jbang.spi.IntegrationManager.runIntegrations(IntegrationManager.java:81)
	... 16 more
[jbang] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

--- a/.github/workflows/ci-build.yml
+++ b/.github/workflows/ci-build.yml
@@ -114,7 +114,7 @@ jobs:
     needs: build
     strategy:
       ## disable fail-fast so on variance failing don't fail the rest
-      fail-fast: false
+      #fail-fast: false
       matrix:
         os: [ubuntu-latest]
         java: [8,11,17]
@@ -151,7 +151,7 @@ jobs:
           rm qcli.java
       - name: integration-test-non-windows
         ## karate can't run on java 17 or higher due to missing nashorn
-        if: runner.os != 'Windows' || runner.java != '17'
+        if: runner.os != 'Windows' && matrix.java < 17
         run: |
           export PATH=`pwd`/build/install/jbang/bin:$PATH
           cd build/itests
@@ -162,7 +162,7 @@ jobs:
       - name: add-to-path
         run: echo "../build/install/jbang/bin" >> $env:GITHUB_PATH
       - name: integration-test-windows
-        if: runner.os == 'Windows'
+        if: runner.os == 'Windows' && matrix.java < 17
         working-directory: itests
         run: |
           echo "::add-matcher::.github/karatewindows-matcher.json" 

--- a/build.gradle
+++ b/build.gradle
@@ -31,7 +31,7 @@ dependencies {
 	implementation 'io.quarkus.qute:qute-core:1.12.2.Final'
 	implementation 'kr.motd.maven:os-maven-plugin:1.7.0'
 	implementation 'org.codehaus.plexus:plexus-java:1.0.6'
-	implementation 'com.google.code.gson:gson:2.8.6'
+	implementation 'com.google.code.gson:gson:2.9.0'
 	implementation 'org.jsoup:jsoup:1.13.1'
 	implementation 'org.codejive:java-properties:0.0.4'
 

--- a/src/main/java/dev/jbang/spi/IntegrationManager.java
+++ b/src/main/java/dev/jbang/spi/IntegrationManager.java
@@ -1,9 +1,21 @@
 package dev.jbang.spi;
 
-import static dev.jbang.cli.BaseCommand.*;
+import static dev.jbang.cli.BaseCommand.EXIT_GENERIC_ERROR;
+import static dev.jbang.cli.BaseCommand.EXIT_INVALID_INPUT;
 import static dev.jbang.util.JavaUtil.resolveInJavaHome;
 
-import java.io.*;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileDescriptor;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintStream;
+import java.io.StringReader;
+import java.io.Writer;
 import java.lang.reflect.Method;
 import java.net.MalformedURLException;
 import java.net.URL;
@@ -12,7 +24,13 @@ import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 import javax.annotation.Nonnull;
@@ -40,7 +58,8 @@ public class IntegrationManager {
 	public static final String MAIN_CLASS = "main-class";
 	public static final String JAVA_ARGS = "java-args";
 
-	private static final GsonBuilder gsonb = new GsonBuilder().registerTypeAdapter(Path.class, new PathTypeAdapter());
+	private static final GsonBuilder gsonb = new GsonBuilder().registerTypeHierarchyAdapter(Path.class,
+			new PathTypeAdapter());
 
 	/**
 	 * Discovers all integration points and runs them.

